src/main/java/studio/magemonkey/genesis/managers/item/ItemDataPartArmorTrim.java

```diff
@@ -0,0 +1,80 @@
+package studio.magemonkey.genesis.managers.item;
+
+import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ArmorMeta;
+import org.bukkit.inventory.meta.trim.ArmorTrim;
+import org.bukkit.inventory.meta.trim.TrimMaterial;
+import org.bukkit.inventory.meta.trim.TrimPattern;
+import studio.magemonkey.genesis.core.GenesisBuy;
+import studio.magemonkey.genesis.managers.ClassManager;
+
+import java.util.List;
+
+public class ItemDataPartArmorTrim extends ItemDataPart{
+    @Override
+    public ItemStack transform(ItemStack item, String usedName, String argument) {
+        String[] parts = argument.split("#");
+        if (parts.length != 2) {
+            ClassManager.manager.getBugFinder()
+                    .severe("Mistake in Config: '" + argument + "' is not a valid '" + usedName
+                            + "'. It has to look like this: '<trim-pattern>#<trim-material>'.");
+            return item;
+        }
+
+        TrimPattern trimPattern = Registry.TRIM_PATTERN.get(NamespacedKey.minecraft(parts[0]));
+        if (trimPattern == null) {
+            ClassManager.manager.getBugFinder().severe("Mistake in Config: '" + parts[0] + "' is not a valid '" + usedName + "' pattern!");
+            return item;
+        }
+        TrimMaterial trimMaterial = Registry.TRIM_MATERIAL.get(NamespacedKey.minecraft(parts[1]));;
+        if (trimMaterial == null) {
+            trimMaterial = TrimMaterial.IRON;
+            ClassManager.manager.getBugFinder().severe("Mistake in Config: '" + parts[1] + "' is not a valid '" + usedName + "' material!");
+        }
+        if (item.getItemMeta() instanceof ArmorMeta) {
+            ArmorTrim armorTrim = new ArmorTrim(trimMaterial, trimPattern);
+            ArmorMeta itemMeta = (ArmorMeta) item.getItemMeta();
+            itemMeta.setTrim(armorTrim);
+            item.setItemMeta(itemMeta);
+            return item;
+        }
+        ClassManager.manager.getBugFinder().severe("Mistake in Config: Unable to add '" + usedName + "' to '" + item.getType().name() + "'!");
+        return item;
+    }
+
+    @Override
+    public boolean isSimilar(ItemStack shopItem, ItemStack playerItem, GenesisBuy buy, Player p) {
+        return false;
```

Comment:
Only thing I'm concerned about is that this isn't set up to make sure that the trim matches.

Reply:
I've added functionality to the method as per request.